Src/Base/AMReX_GpuLaunchFunctsG.H

```diff
@@ -747,35 +747,76 @@ void launch (int nblocks, int nthreads_per_block, gpuStream_t stream, L&& f) noe
     launch(nblocks, nthreads_per_block, 0, stream, std::forward<L>(f));
 }
 
-template<int MT, typename T, typename L>
+template<int MT, typename T, typename L, std::enable_if_t<std::is_integral_v<T>,int> FOO = 0>
 void launch (T const& n, L const& f) noexcept
 {
+    static_assert(sizeof(T) >= 2);
     if (amrex::isEmpty(n)) { return; }
-    const auto ec = Gpu::makeExecutionConfig<MT>(n);
-    AMREX_LAUNCH_KERNEL(MT, ec.numBlocks, ec.numThreads, 0, Gpu::gpuStream(),
-    [=] AMREX_GPU_DEVICE () noexcept {
-        for (auto const i : Gpu::Range(n)) {
-            f(i);
+    const auto& nec = Gpu::makeNExecutionConfigs<MT>(n);
+    T ndone = 0;
+    for (auto const& ec : nec) {
+        T nleft = n - ndone;
+        AMREX_LAUNCH_KERNEL(MT, ec.nblocks, MT, 0, Gpu::gpuStream(),
+        [=] AMREX_GPU_DEVICE () noexcept {
+            // This will not overflow, even though nblocks*MT might.
+            auto tid = T(MT)*T(blockIdx.x)+T(threadIdx.x);
+            if (tid < nleft) {
+                f(tid+ndone);
+            }
+        });
+        if (Long(nleft) > ec.ntotalthreads) {
+            ndone += T(ec.ntotalthreads);
```

Comment:
Maybe ndone and nleft can be part of ExecConfig to simplify this here.

Reply:
But ndone and nleft need to be captured by the device lambda function. So I am not sure how to simplify this in a clear way.

Reply:
```
const T ndone = T(ec.ndone);
const T nleft = T(ec.nleft);
```

Reply:
But I am not sure what advantage it has. We have to make `ec.ndone` and `ec.nleft` static members of `nec`, which current is Vector. We will have to define our own types and support range for loop.

Reply:
I added a suggested change to show what I mean